lua/chatcommands/duel.lua

```diff
@@ -0,0 +1,313 @@
+        return
+    elseif (findDuel(duel.challenger)) then
+        target:ChatPrint("[red]The player you are trying to fight already has a active duel.")
+        return
+    elseif (not duel.challenger:SS_HasPoints(duel.points)) then
+        duel.challenger:ChatPrint("[red]You do not have enough points, duel was cancelled.")
+        duel.target:ChatPrint("[red]The challenger does not have enough points, duel was cancelled.")
+        DuelRequests[duelIndex] = nil
+        return
+    elseif (not duel.target:SS_HasPoints(duel.points)) then
+        duel.target:ChatPrint("[red]You do not have enough points to accept this duel.")
+        duel.challenger:ChatPrint("[red]The challenged player does not have enough points, duel was cancelled.")
+        DuelRequests[duelIndex] = nil
+        return
+    end
+
+    duel.challenger:ChatPrint("[orange]" .. duel.target_name .. " accepted your duel request, duel will start in 20 seconds!")
+    duel.target:ChatPrint("[orange]Your duel with " .. duel.challenger_name .. " will start in 20 seconds!")
+
+    duel.challenger:SS_TakePoints(duel.points)
+    duel.target:SS_TakePoints(duel.points)
+
+    table.insert(AcceptedDuels, {challenger = duel.challenger, target = duel.target, challenger_name = duel.challenger_name, target_name = duel.target_name, points = duel.points, kills = duel.kills})
+    DuelRequests[duelIndex] = nil;
+
+    startCountdown(#AcceptedDuels)
+end
+
+function startCountdown(duelIndex)
+    local duel = AcceptedDuels[duelIndex]
+    local time = 20
+    timer.Create( "CountdownDuel" .. duelIndex, 1, 20, function()
+        time = time - 1
+
+        if (not IsValid(duel.challenger)) then
+            duel.target:SS_GivePoints(duel.points * 2)
+            duel.target:ChatPrint("[red]Your opponent " .. duel.challenger_name .. " has left the game. [green]You won [gold]" .. string.Comma(duel.points) .. "[green] points.")
+            BotSayGlobal("[fbc]" .. duel.target_name .. " won a duel worth [rainbow]" .. string.Comma(duel.points * 2) .. " points [fbc] against [gold]" .. duel.challenger_name .. "[fbc]. Their opponent left the game.")
+            AcceptedDuels[duelIndex] = nil
+            timer.Remove( "CountdownDuel" .. duelIndex )
+            return
+        elseif (not IsValid(duel.target)) then
+            duel.challenger:SS_GivePoints(duel.points * 2)
+            duel.challenger:ChatPrint("[red]Your opponent " .. duel.target_name .. " has left the game. [green]You won [gold]" .. string.Comma(duel.points) .. "[green] points.")
+            BotSayGlobal("[fbc]" .. duel.challenger_name .. " won a duel worth [rainbow]" .. string.Comma(duel.points * 2) .. " points [fbc] against [gold]" .. duel.target_name .. "[fbc]. Their opponent left the game.")
+            AcceptedDuels[duelIndex] = nil
+            timer.Remove( "CountdownDuel" .. duelIndex )
+            return
+        end
+
+        if (time == 15) then
```

Comment:
make this shit less redundant 

```
if ({15=true,10=true,5=true,4=true,3=true,2=true,1=true})[time] then
    for i,ply in ipairs({duel.challenger,duel.target}) do
        ply:ChatPrint("[orange]Duel starts in " .. time .. " seconds")
    end
end
```

Reply:
all the other places you do the same thing for both players should also loop over them in the same way

Reply:
Well I learned something today, I didn't know about the bracket thing in if statements